source/opt/freeze_spec_constant_value_pass.cpp

```diff
@@ -49,5 +60,115 @@ Pass::Status FreezeSpecConstantValuePass::Process() {
   return modified ? Status::SuccessWithChange : Status::SuccessWithoutChange;
 }
 
+Pass::Status FreezeSpecConstantValuePass::FreezeSpecIds(
+    const SpecIdSet& spec_ids) {
+  // The operand index of decoration target in an OpDecorate instruction.
+  const uint32_t kTargetIdOperandIndex = 0;
+  // The operand index of the decoration literal in an OpDecorate instruction.
+  const uint32_t kDecorationOperandIndex = 1;
+  // The operand index of Spec id literal value in an OpDecorate SpecId
+  // instruction.
+  const uint32_t kSpecIdLiteralOperandIndex = 2;
+  // The number of operands in an OpDecorate SpecId instruction.
+  const uint32_t kOpDecorateSpecIdNumOperands = 3;
+
+  auto ctx = context();
+  auto def = get_def_use_mgr();
```

Comment:
please name this `def_use_mgr` or something like that.  The name `def` make it sound like it is the definition of a value.